provider/testdata/data/baked/mod.rs

```diff
@@ -20,2087 +20,2890 @@ mod time_zone;
-                *collator::meta_v1::DATA
-                    .get_by(|k| req.locale.strict_cmp(k.as_bytes()).reverse())
-                    .ok_or_else(|| DataErrorKind::MissingLocale.with_req(::icu_collator::provider::CollationMetadataV1Marker::KEY, req))?,
-            ))),
-        })
-    }
-}
-#[cfg(feature = "icu_collator")]
-impl DataProvider<::icu_collator::provider::CollationReorderingV1Marker> for BakedDataProvider {
-    fn load(&self, req: DataRequest) -> Result<DataResponse<::icu_collator::provider::CollationReorderingV1Marker>, DataError> {
-        Ok(DataResponse {
-            metadata: Default::default(),
-            payload: Some(DataPayload::from_owned(zerofrom::ZeroFrom::zero_from(
-                *collator::reord_v1::DATA
-                    .get_by(|k| req.locale.strict_cmp(k.as_bytes()).reverse())
-                    .ok_or_else(|| DataErrorKind::MissingLocale.with_req(::icu_collator::provider::CollationReorderingV1Marker::KEY, req))?,
-            ))),
-        })
-    }
-}
-#[cfg(feature = "icu_collator")]
-impl DataProvider<::icu_collator::provider::CollationSpecialPrimariesV1Marker> for BakedDataProvider {
-    fn load(&self, req: DataRequest) -> Result<DataResponse<::icu_collator::provider::CollationSpecialPrimariesV1Marker>, DataError> {
-        Ok(DataResponse {
-            metadata: Default::default(),
-            payload: Some(DataPayload::from_owned(zerofrom::ZeroFrom::zero_from(
-                *collator::prim_v1::DATA
-                    .get_by(|k| req.locale.strict_cmp(k.as_bytes()).reverse())
-                    .ok_or_else(|| DataErrorKind::MissingLocale.with_req(::icu_collator::provider::CollationSpecialPrimariesV1Marker::KEY, req))?,
-            ))),
-        })
-    }
-}
-#[cfg(feature = "icu_compactdecimal")]
-impl DataProvider<::icu_compactdecimal::provider::LongCompactDecimalFormatDataV1Marker> for BakedDataProvider {
-    fn load(&self, req: DataRequest) -> Result<DataResponse<::icu_compactdecimal::provider::LongCompactDecimalFormatDataV1Marker>, DataError> {
-        Ok(DataResponse {
-            metadata: Default::default(),
-            payload: Some(DataPayload::from_owned(zerofrom::ZeroFrom::zero_from(
-                *compactdecimal::long_v1::DATA
-                    .get_by(|k| req.locale.strict_cmp(k.as_bytes()).reverse())
+/// Uses baked data to implement DataProvider<M> on the given struct.
+/// This macro can only be called from its definition-site, i.e. right
+/// after `include!`-ing the baked file.
+macro_rules! implement_baked_provider {
+    ($ provider : path) => {
+        #[cfg(feature = "icu_calendar")]
+        impl DataProvider<::icu_calendar::provider::JapaneseErasV1Marker> for $provider {
+            fn load(&self, req: DataRequest) -> Result<DataResponse<::icu_calendar::provider::JapaneseErasV1Marker>, DataError> {
+                calendar::japanese_v1::lookup(&req.locale)
+                    .map(zerofrom::ZeroFrom::zero_from)
```

Comment:
Thought: Although `ZeroFrom` doesn't allocate (except for the skeleton hack), it's not completely free, and it shows up in code size benches. We may want to think about making DataPayload copy-on-write where it can retain a reference to the static data that it returns from `.get()` and only call `ZeroFrom` when mutation needs to occur (rarer case).

Reply:
I'd love to but Yoke doesn't currently support this afaik. @Manishearth

Reply:
It doesn't need to be in Yoke; I think it can be in DataPayload. (Not now; maybe for a future PR)